ynr/apps/sopn_parsing/management/commands/sopn_tooling_create_official_documents.py

```diff
@@ -0,0 +1,72 @@
+import requests
+
+from django.core.management.base import BaseCommand
+from django.core.files.base import ContentFile
+
+from candidates.models import Ballot
+from official_documents.models import OfficialDocument
+
+
+class Command(BaseCommand):
+    """This command uses the ballots endpoint to loop over each
+    ballot and store each sopn pdf (uploaded_file) locally"""
+
+    def handle(self, *args, **options):
+        response = requests.get(
+            "https://candidates.democracyclub.org.uk/api/next/ballots?has_sopn=1&page_size=20"
+        )
+        data = response.json()
+        self.create_official_documents(ballots=data["results"])
+
+    def create_official_documents(self, ballots):
+        for ballot_data in ballots:
+            ballot = Ballot.objects.get(
```

Comment:
The management command errors for me here: 
`Ballot matching query does not exist.` Should we add a conditional to check for the existence of the ballot? 

If the `ballot_paper_id` exists in the `balllot_data`, but not in the Ballot Query, what action should be taken?

Reply:
On second thought, why else would a Ballot query not work here? I tried hardcoding the ballot_paper_id and got the same error.

Reply:
Ah something I forgot to consider - you'll need an up-to-date database, that will be why you are getting that error. Use the `candidates_import_from_live_site` management command to do that

Reply:
Or this could be updated to use a date range to limit the ballots used. I'll add that to my TODO list as it would mean that we would be able to then always test the same set of SOPN's

Reply:
@VirginiaDooley I've just pushed a commit that adds some defaults to the URL about which ballots to use - defaulting to elections from May 6 this year. You should have these in your database so you should not face the `BallotDoesNotExist` error

Reply:
Original steps no longer relevant - see the updated README for steps to test locally. Comments on the need for any additional clarity/instructions etc useful!